ksqldb-rest-app/src/test/java/io/confluent/ksql/api/server/OldApiUtilsTest.java

```diff
@@ -0,0 +1,111 @@
+
+import static io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.rest.EndpointResponse;
+import io.confluent.ksql.rest.Errors;
+import io.confluent.ksql.rest.entity.KsqlErrorMessage;
+import io.confluent.ksql.rest.entity.KsqlRequest;
+import io.confluent.ksql.rest.server.resources.KsqlRestException;
+import io.confluent.ksql.util.VertxCompletableFuture;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.ext.web.RoutingContext;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class OldApiUtilsTest {
+
+  @Mock
+  private Server server;
+  @Mock
+  private HttpServerRequest request;
+  @Mock
+  private RoutingContext routingContext;
+
+  @Before
+  public void setUp() {
+    when(routingContext.request()).thenReturn(request);
+  }
+
+  @Test
+  public void shouldMaskKsqlRequestQuery() throws ExecutionException, InterruptedException {
+    // Given
+    final String query = "--this is a comment. \n"
+        + "CREATE SOURCE CONNECTOR `test-connector` WITH ("
+        + "    \"connector.class\" = 'PostgresSource', \n"
+        + "    'connection.url' = 'jdbc:postgresql://localhost:5432/my.db',\n"
+        + "    \"mode\"='bulk',\n"
+        + "    \"topic.prefix\"='jdbc-',\n"
+        + "    \"table.whitelist\"='users',\n"
+        + "    \"key\"='username');";
+    final KsqlRequest req = new KsqlRequest(query, ImmutableMap.of(), ImmutableMap.of(), 1L);
+    when(routingContext.getBody()).thenReturn(ServerUtils.serializeObject(req));
+
+    // when, Then
+    final String expected = "CREATE SOURCE CONNECTOR `test-connector` WITH "
+        + "(\"connector.class\"='PostgresSource', "
+        + "'connection.url'='[string]', "
+        + "\"mode\"='[string]', "
+        + "\"topic.prefix\"='[string]', "
+        + "\"table.whitelist\"='[string]', "
+        + "\"key\"='[string]');";
+
+    OldApiUtils.handleOldApiRequest(server, routingContext, KsqlRequest.class, Optional.empty(),
+        (ksqlRequest, securityContext) -> {
+      assertThat(ksqlRequest.getMaskedKsql(), is(expected));
+      final VertxCompletableFuture<EndpointResponse> vcf = new VertxCompletableFuture<>();
+      vcf.complete(EndpointResponse.failed(1));
+      return vcf;
+    });
+  }
+
+  @Test
+  public void shouldReturnEmbeddedResponseForKsqlRestException() {
```

Comment:
Delete the other test file of the name same, if we're going to move these tests into this one.